graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/IndexSetsResource.java

```diff
@@ -154,6 +154,29 @@ public IndexSetResponse list(@ApiParam(name = "skip", value = "The number of ele
     }
 
     @GET
+    @Path("stats")
+    @Timed
+    @ApiOperation(value = "Get stats of all index sets")
+    @ApiResponses(value = {
+            @ApiResponse(code = 403, message = "Unauthorized"),
+    })
+    public IndexSetStats globalStats() {
+        checkPermission(RestPermissions.INDEXSETS_READ);
+
+        long indices = 0;
+        long documents = 0;
+        long size = 0;
+        for (IndexSetStats stats : indexSetRegistry.getAll().stream()
+                .collect(Collectors.toMap(indexSet -> indexSet.getConfig().id(), indexSetStatsCreator::getForIndexSet)).values()) {
```

Comment:
Can you please extract this (the map you are iterating over) into a local variable? This makes it way easier to understand what is going on and the type is explicitly visible.

Reply:
Also, why collect into a map if you're only interested in the values anyway?

Reply:
Additionally, this will run two remote queries for each index set for a piece of information which could be fetched with a single request for all index sets via the [Cat Indices API](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/cat-indices.html).